packages/api/src/router/user.ts

```diff
@@ -15,6 +20,29 @@ export const userRouter = createTRPCRouter({
     return new Map(users.map((user) => [user.id, user]));
   }),
 
+  /**
+   * Checks if the user with the given email is approved in clerk.
+   * Since we want this to be callable when the user is not authenticated, it must
+   * remain public. Thus, we should not return sensitive data and only include the
+   * bare minimum in the return value.
+   * @returns an object of the form {userExists: bool, isApproved: bool}
+   */
+  isEmailApproved: publicProcedure
+    .input(z.string())
+    .query(async ({ input }) => {
+      const users = await clerkClient.users.getUserList({
```

Comment:
use getUser here instead

Reply:
getUser only has filter by `userId`, not the email

Reply:
🤔 instead of storing the isApproved stuff in clerk, can we just make it so that only approved users exist in the db. Then you only need to check the db here?

Reply:
So we'd need another table in the DB storing approved emails? 

Maybe `isApproved` should be a field in our DB instead of Clerk, and we insert everyone into the DB, but default their approval to `false`. The admin dashboard can set the approval status within our db (so clerk would store no metadata)

Reply:
I don't think we would need another table -- but yea adding the isApproved into our dashboard instead. I think that would work well and should be an easy change